pkg/api/v20170701/types.go

```diff
@@ -50,31 +64,31 @@ type Properties struct {
 //    <NAME> is the name of the secret.
 //    <VERSION> (optional) is the version of the secret (default: the latest version)
 type ServicePrincipalProfile struct {
-	ClientID string `json:"clientId,omitempty"`
-	Secret   string `json:"secret,omitempty"`
+	ClientID string `json:"clientId,omitempty" validate:"required"`
+	Secret   string `json:"secret,omitempty" validate:"required"`
 }
 
 // CustomProfile specifies custom properties that are used for
 // cluster instantiation.  Should not be used by most users.
 type CustomProfile struct {
-	Orchestrator string `json:"orchestrator,omitempty"`
+	Orchestrator string `json:"orchestrator,omitempty" validate:"required"`
```

Comment:
Take a closer look, CustomProfile doesn't have any validate function. I don't want to add required here anymore.